docs/src/background/infeasibility_certificates.md

```diff
@@ -0,0 +1,113 @@
+```@meta
+CurrentModule = MathOptInterface
+DocTestSetup = quote
+    using MathOptInterface
+    const MOI = MathOptInterface
+end
+DocTestFilters = [r"MathOptInterface|MOI"]
+```
+
+# Infeasibility certificates
+
+When given an infeasible or unbounded model, some (conic) solvers can produce a
+certificate or infeasibility. This page explains what a certificate of
+infeasibility is for unbounded and infeasible models.
+
+## Conic duality
+
+MathOptInterface uses conic duality, which we use to define infeasibility
+certificates. A full explanation is given in the section [Duality](@ref), but
+here is a brief overview.
+
+For a minimization problem in geometric conic form, the primal is:
+```math
+\begin{align}
+& \min_{x \in \mathbb{R}^n} & a_0^\top x + b_0
+\\
+& \;\;\text{s.t.} & A_i x + b_i & \in \mathcal{C}_i & i = 1 \ldots m
+\end{align}
+```
+and the dual is a maximization problem in standard conic form:
+```math
+\begin{align}
+& \max_{y_1, \ldots, y_m} & -\sum_{i=1}^m b_i^\top y_i + b_0
+\\
+& \;\;\text{s.t.} & a_0 - \sum_{i=1}^m A_i^\top y_i & = 0
+\\
+& & y_i & \in \mathcal{C}_i^* & i = 1 \ldots m
+\end{align}
+```
+where each ``\mathcal{C}_i`` is a closed convex cone and ``\mathcal{C}_i^*`` is
+its dual cone.
+
+For a maximization problem in geometric conic form, the primal is:
+```math
+\begin{align}
+& \max_{x \in \mathbb{R}^n} & a_0^\top x + b_0
+\\
+& \;\;\text{s.t.} & A_i x + b_i & \in \mathcal{C}_i & i = 1 \ldots m
+\end{align}
+```
+and the dual is a minimization problem in standard conic form:
+```math
+\begin{align}
+& \min_{y_1, \ldots, y_m} & \sum_{i=1}^m b_i^\top y_i + b_0
+\\
+& \;\;\text{s.t.} & a_0 + \sum_{i=1}^m A_i^\top y_i & = 0
+\\
+& & y_i & \in \mathcal{C}_i^* & i = 1 \ldots m
+\end{align}
+```
+
+## Unbounded models
+
+For an unbounded model, a certificate of infeasibility is an improving ray of
```

Comment:
"or an unbounded model, a certificate of infeasibility" can be confusing as it's not clear whether we're certifying unboundness or infeasiblity. I would add something like that:

For an unbounded model, a certificate of unboundedness is made of two separate certificates:
1) A feasible solution `x0` certifying that the primal is feasible
2) A direction `d` that is both feasible for the homogenized feasible set and is a direction of decrease for the objective (in case of minimization).

Combining these two certificates, by convexity of the feasible set and linearity of the objective function, we can see that `x0 + \eta d` is feasible for any \eta >= 0 and can have an objective value arbitrarily small for increasing values of \eta.

The direction `d` is also a certificate of infeasibility of the dual problem.
In fact, most solvers stop when they have found `d` and do not search for a feasible solution `x0` so they terminate with a complete certificate of dual infeasibility but an incomplete certificate of primal unboundedness.
In that situation, the dual is known to be infeasible but the primal can still be either infeasible or unbounded.
The two cases can be distinguished by setting the `ObjectiveSense` to `FEASIBILITY_SENSE` and calling `optimize!` on the modified model.

Reply:
Yeah I should just call it primal and dual infeasibility certificates and add a note about solving with no objective.